### PR TITLE
WhiteSource scan for LTS branch

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

**Comment by vilchik-elena** [comment]
@saberduck can we close this PR?

**Comment by saberduck** [comment]
@vilchik-elena yes, we do. let's discuss tomorrow how to fix this :)

File: .cirrus.yml [hunk 1]
```diff
@@ -2,17 +2,18 @@ gcp_credentials: ENCRYPTED[!17c59813193e86cc337bce848b358412b90f50bc5fe1b8b39d36
...
   ARTIFACTORY_DEPLOY_REPO: sonarsource-public-qa
-  ARTIFACTORY_API_KEY: ENCRYPTED[!35ca4446564213d4fd2d1a96e42a871d5de6e6aac4e1dd3e89342892a60a2badf74a966bcc8e885e9c9d09a775ffe4c0!]
+  ARTIFACTORY_API_KEY: ENCRYPTED[!815fc7526a6579b7d974d1c41b5c247cfed8c88fb2c7b41ec5d3d54244701186da777c4003476b5b9fa208c8b8f208dc!]
   # burgr notification
   BURGR_URL: ENCRYPTED[!c7e294da94762d7bac144abef6310c5db300c95979daed4454ca977776bfd5edeb557e1237e3aa8ed722336243af2d78!]
   BURGR_USERNAME: ENCRYPTED[!b29ddc7610116de511e74bec9a93ad9b8a20ac217a0852e94a96d0066e6e822b95e7bc1fe152afb707f16b70605fddd3!]
   BURGR_PASSWORD: ENCRYPTED[!83e130718e92b8c9de7c5226355f730e55fb46e45869149a9223e724bb99656878ef9684c5f8cfef434aa716e87f4cf2!]
-
+  #npm config with repox npm registry and repox-private-reader-e2882e auth
+  NPMRC: ENCRYPTED[!ef57edd224eb5867d50bdaa3c8570f8efb944aea8f4edbdab6be2a14355bf8257718da0a84e8f6fa3aa028a67de9d921!]
```
Comment (vilchik-elena) [inline .cirrus.yml:16]: why we need authorization to read from repox?

**Review by vilchik-elena — CHANGES_REQUESTED** [review]
Will this CI run daily?

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/EslintBridgeIntegrationTest.java [hunk 2]
```diff
@@ -99,8 +100,10 @@ private void assertAnalyzeJs(EslintBridge eslintBridge) throws IOException {
     JsonArray issues = jsonObject.getAsJsonArray("issues");
     assertThat(issues).hasSize(3);
     assertThat(issues).extracting(i -> i.getAsJsonObject().get("line").getAsInt()).containsExactlyInAnyOrder(2, 3, 4);
-    // this assert makes sure that we don't compute metrics
-    assertThat(jsonObject.entrySet()).extracting(Map.Entry::getKey).containsOnly("issues");
+    // this assert makes sure that we don't compute metrics except nosonar lines
+    JsonObject metrics = jsonObject.getAsJsonObject("metrics");
+    assertThat(metrics.entrySet()).hasSize(1);
+    assertThat(metrics.get("nosonarLines").getAsJsonArray()).containsExactly(new JsonPrimitive(3));
```
[the remark this answers is not in the document]
Comment (saberduck) [reply its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/EslintBridgeIntegrationTest.java:106]: no sonar is needed in sonarlint

File: eslint-bridge/yarn.lock [hunk 3]
```diff
@@ -1846,10 +1846,10 @@ eslint-plugin-chai-friendly@0.6.0:
   resolved "https://registry.yarnpkg.com/eslint-plugin-chai-friendly/-/eslint-plugin-chai-friendly-0.6.0.tgz#54052fab79302ed0cea76ab997351ea4809bfb77"
   integrity sha512-Uvvv1gkbRGp/qfN15B0kQyQWg+oFA8buDSqrwmW3egNSk/FpqH2MjQqKOuKwmEL6w4QIQrIjDp+gg6kGGmD3oQ==
 
-eslint-plugin-sonarjs@0.7.0:
-  version "0.7.0"
-  resolved "https://registry.yarnpkg.com/eslint-plugin-sonarjs/-/eslint-plugin-sonarjs-0.7.0.tgz#2dbad45e03a6a4de687756b659e88bf3f977383f"
-  integrity sha512-vi6zGkd5Eznc32AQlleWUOMrMeDiUih9wR7nPPfrDCyVRmwYNHIBRPZGv1EgXwELwaPghCSvoAoHoR7uSbBF/Q==
+eslint-plugin-sonarjs@0.7.1-102:
```
[the remark this answers is not in the document]
Comment (saberduck) [reply eslint-bridge/yarn.lock:1849]: not sure, I will try revert

File: .cirrus.yml [hunk 4]
```diff
@@ -57,9 +58,11 @@ build_task:
     PGP_PASSPHRASE: ENCRYPTED[!314a8fc344f45e462dd5e8dccd741d7562283a825e78ebca27d4ae9db8e65ce618e7f6aece386b2782a5abe5171467bd!]
   maven_cache:
     folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  npm_config_file:
+    path: ~/.npmrc
+    variable_name: NPMRC
   build_script:
-    - source cirrus-env BUILD
-    - yarn config set registry "${ARTIFACTORY_URL}/api/npm/npm"
```
[the remark this answers is not in the document]
Comment (saberduck) [reply .cirrus.yml:62]: no idea, but it works?

File: .cirrus.yml (hunk 1 above)
Comment (saberduck) [reply .cirrus.yml:16]: I believe this is just standard setup they apply to everything

**Review by vilchik-elena — APPROVED** [review]

File: eslint-bridge/yarn.lock (hunk 3 above)
[the remark this answers is not in the document]
Comment (saberduck) [reply eslint-bridge/yarn.lock:1849]: this is to match what's in package.json

**Pull request merged**